### PR TITLE
refactor: use new deno_ast emitter API

### PR DESCRIPTION
**Pull request by lucacasonato**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -59,3 +59,6 @@ codegen-units = 1
 incremental = true
 lto = true
 opt-level = "z"
+
+[patch.crates-io]
+deno_ast = { path = "../deno_ast" }
```
Comment (marvinhagemeister) [inline Cargo.toml:64]: This was probably accidentally committed.

Comment (dsherret) [reply Cargo.toml:64]: It's just temporary until the deno_ast changes land: https://github.com/denoland/deno_ast/pull/230

File: tests/specs/graph/jsr/FastCheck_ClassMethodsWithTsPrivate.txt [hunk 2]
```diff
@@ -157,7 +157,7 @@ Fast check https://jsr.io/@scope/a/1.0.0/mod.ts:
     private "stringOnly"(value: string): string;
     private "stringOnly"(): string;
   }
-  export declare namespace Test {
+  export declare module Test {
```
Comment (dsherret) [inline tests/specs/graph/jsr/FastCheck_ClassMethodsWithTsPrivate.txt:160]: Due to swc downgrade

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**